### ISSUE TITLE
How to generate MSRP URI for using Main.java test program

### ISSUE THREAD

**Issue by stevesui** [archiecobbs/msrp4j#1]
I downloaded and compiled MSRP4J version 2.0.0 and tried to run the Main test program.
from msrp4j-> console, tried to use "connect localURI remoteURI" option. I am not sure how to generate localURI and remoteURI. 

I thought the running program (Main.java) should generate its own local MsrpURI. Obviously it did not.

Can someone help to explain how the MsrpURI is generated for running the test pragram?

Appreciate it very much.

**Comment by stevesui** [comment]
To follow up my previous issue. In the command line, to construct a msrp uri, how should I fill out the transaction id part? On command line, If I give: "connect  msrp://localhost:2856/abc;tcp localhost:2855/abc?tcp", the transaction id part "abc" is what I hard-coded there,  not a valid id generated from MSRP Session though.  And It was rejected. How could I generate a correct msrp uri without session telling me the transaction id?

**Comment by archiecobbs** [comment]
MSRP URI's are generated by whatever higher layer you are using to negotiate the protocol. Usually this is SIP. See [RFC 4975](https://tools.ietf.org/html/rfc4975) for more info.

This works for me:

Session 1:
```
$ java -jar msrp4j-2.0.0-cli.jar --port 3002
Welcome to msrp4j. Type `help' for help.
Listening on address 0.0.0.0/0.0.0.0:3002
msrp4j> accept msrp://localhost:3002/aaaa;tcp msrp://localhost:3001/bbbb;tcp
* Created session: local=msrp://localhost:3002/aaaa;tcp remote=msrp://localhost:3001/bbbb;tcp
* New current session is msrp://localhost:3002/aaaa;tcp
msrp4j> 
```

Session 2:

```
$ java -jar msrp4j-2.0.0-cli.jar --port 3001
Welcome to msrp4j. Type `help' for help.
Listening on address 0.0.0.0/0.0.0.0:3001
msrp4j> connect msrp://localhost:3001/bbbb;tcp  msrp://localhost:3002/aaaa;tcp 
* Created session: local=msrp://localhost:3001/bbbb;tcp remote=msrp://localhost:3002/aaaa;tcp
* New current session is msrp://localhost:3001/bbbb;tcp
msrp4j> text hello
* Sent message with message ID 22e90b703581d61c
msrp4j> 
```

Session 1:
```
* Rec'd message from msrp://localhost:3001/bbbb;tcp:
  Message-ID: 55ea05469389b1bb
  Success-Report: no
  Failure-Report: no
  [ Message contains no content ]
* Rec'd message from msrp://localhost:3001/bbbb;tcp:
  Message-ID: 22e90b703581d61c
  Success-Report: yes
  Failure-Report: yes
  Content-Type: text/plain; charset=utf-8
  Actual Length: 5 bytes
  Content SHA1: aaf4c61ddcc5e8a2dabede0f3b482cd9aea9434d

hello

msrp4j> 
```

**Comment by stevesui** [comment]
Archie:
Thanks a lot for the quick reply. I figured it out after I posted the issue. I surely understand SIP INVITE is involved for handshaking before MSRP session established. Here is what I did:

I ran the msrp receiver instance on port 2855 and  run "accept" first with generated transaction id (abc), msrp->"accept msrp://localhost:2855/abc;tcp msrp://localhost:2856/abc;tcp"; 

I ran msrp sender instance on port 2856 and run msrp->"connect msrp://127.0.0.1:2856/abc;tcp msrp://localhost:2855/abc;tcp. 

After that, both side established msrp session.  

Next, I can send messages to each other using msrp->text "messages". 

Again thanks a lot.

Steve
